web/packages/teleport/src/services/joinToken/types.ts

```diff
@@ -117,32 +117,32 @@ export type OracleRules = {
 
 type GithubConfig = {
   /** enterprise_server_host allows joining from GitHub Actions workflows in a GitHub Enterprise Server instance. For normal situations, where you are using github.com, this option should be omitted. If you are using GHES, this value should be configured to the hostname of your GHES instance. */
-  enterprise_server_host: string | null;
+  enterprise_server_host?: string | null;
```

Comment:
What's the reason behind preferring this over `enterprise_server_host?: string`? Do these have to align with values serialized from Go?

I feel like on paper optional fields that also can be null are an unusual concept which the community came to accept only because it's so popular in JS.

Something like protobuf-ts that is strict mode compatible generates optional keys but non-nullable values, e.g.:

https://github.com/gravitational/teleport/blob/bc5755f00f77a9f44a6774b2bcdb6b8d030a0114/gen/proto/ts/teleport/lib/teleterm/v1/gateway_pb.ts#L106

Not a blocker, I just wonder if long-term we should follow either approach.